04-onion-routing.md

```diff
@@ -1068,6 +1178,21 @@ key, and computes the HMAC, using each hop's `um` key.
 The origin node can detect the sender of the error message by matching the
 `hmac` field with the computed HMAC.
 
+For trampoline payments, the flow is the same. Intermediate trampoline nodes
+first decrypt the downstream error using the `ammag` and `um` keys for their
+forward path. If the error comes from an intermediate node _before_ the next
+trampoline node, they may replace it with their own error for the origin node,
```

Comment:
Checking compatibility with attributable failures. This case where the error comes from an intermediate node before the next trampoline node seems fine? The intermediate trampoline decrypts the attributable failure and creates its own with itself as the failure source.

Do trampoline nodes worry about penalties applied to them for failures on the path that they used towards the next trampoline node?

Reply:
That makes sense